### DISCUSSION TITLE
tsx,ts,and js file as page

### DISCUSSION THREAD

**Discussion by hi-reeve** [posva/unplugin-vue-router#117] [Questions and Help] [answered]
right now the plugin is supporting tsx,js,and ts. 
we can add it on `extensions : ['.vue','jsx','tsx','ts','js']`
but the generated `typed-router.d.ts` is including the file name if this setting is enabled.
i.e

```
src/pages/
├── index.vue
├── about.vue
└── users/
    ├── index.vue
    └── [id].vue
```

the route `users/index.vue` is  generated to

```ts
/users/index
```

is this intended or a bug? @posva

**Comment by posva** [comment, marked as answer]
I think you are missing the `.` in front of the extensions. I added a warning and fix during, will push it later.

**Reply by hi-reeve** [reply]
oh right, if i added the `.` it now works normally. thanks!
